helpers/controllers/userInfoController.ts

```diff
@@ -35,21 +35,60 @@ export const userInfo = async (_parent: void, args: UserInfoQueryVariables) => {
     prisma.star.findMany({
       where: {
         mentorId: user.id
+      },
+      include: {
+        student: {
+          select: {
+            username: true,
+            name: true
+          }
+        },
+        lesson: {
+          select: {
+            title: true,
+            order: true
+          }
+        }
+      }
+    }),
+    prisma.star.findMany({
+      where: {
+        studentId: user.id
+      },
+      include: {
+        lesson: {
+          select: {
+            id: true
+          }
+        },
+        mentor: {
+          select: {
+            username: true
+          }
+        }
       }
     })
   ])
 
+  interface lessonMentorMapType {
+    [lessonId: string]: string
+  }
+
+  const lessonMentorMap = starsGiven.reduce((map, starGiven) => {
+    const mentorUsername = _.get(starGiven, 'mentor.username', '')
+    map[starGiven.lessonId!] = mentorUsername
+    return map
+  }, {} as lessonMentorMapType)
+
   const starMap = stars.reduce((map: StarMap, star) => {
-    // TODO change star table so lessonId is not null
-    // the null check is needed so typescript doesn't complain
-    if (!star.lessonId) return map
-    map[star.lessonId] = map[star.lessonId] || []
-    map[star.lessonId].push(star)
+    map[star.lessonId!] = map[star.lessonId!] || []
+    map[star.lessonId!].push(star)
```

Comment:
Remove the not null assertions since they will not be needed anymore.

```suggestion
    map[star.lessonId] = map[star.lessonId] || []
    map[star.lessonId].push(star)
```